app/src/main/java/com/example/caculator/MainActivity.kt

```diff
@@ -12,5 +12,69 @@ class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        updateUI("")
     }
+
+    val opList: MutableList<String> = arrayListOf()
+    val numList: MutableList<String> = arrayListOf()
+
+    fun extractString(item: List<String>, connect: String = ""): String {
+        if (item.isEmpty()) return ""
+        return item.reduce { acc, s -> acc + connect + s }
+    }
+
+    fun updateUI ( mainUIString: String ) {
+        var calculationString = extractString(opList, "")
+        var calculationTxtView = findViewById<View>(R.id.txtResult) as TextView
```

Comment:
In Kotlin, we can get the id of the view without the use of the findViewById syntax. Try it :))